lib/matplotlib/texmanager.py

```diff
@@ -176,8 +176,17 @@ def get_basefile(cls, tex, fontsize, dpi=None):
         Return a filename based on a hash of the string, fontsize, and dpi.
         """
         src = cls._get_tex_source(tex, fontsize) + str(dpi)
-        return os.path.join(
-            cls.texcache, hashlib.md5(src.encode('utf-8')).hexdigest())
+        filehash = hashlib.md5(src.encode('utf-8')).hexdigest()
+
+        nested_folders = Path()
```

Comment:
Why not start this with `cls.texcache` and immediately make `filepath` instead of `nested_folders`?